packages/devextreme/js/__internal/ui/selection/m_selection.ts

```diff
@@ -1,30 +1,40 @@
 import { noop } from '@js/core/utils/common';
-import { Deferred, when } from '@js/core/utils/deferred';
+import { Deferred, type DeferredObj, when } from '@js/core/utils/deferred';
 import { extend } from '@js/core/utils/extend';
-import { isDefined } from '@js/core/utils/type';
-
-import deferredStrategy from './m_selection.strategy.deferred';
-import standardStrategy from './m_selection.strategy.standard';
-
-export default class Selection {
-  options: any;
-
-  _selectionStrategy: deferredStrategy | standardStrategy;
+import { isDefined, isPlainObject } from '@js/core/utils/type';
+import DeferredStrategy from '@ts/ui/selection/m_selection.strategy.deferred';
+import StandardStrategy from '@ts/ui/selection/m_selection.strategy.standard';
+import type {
+  DefaultOptions,
+  SelectionFilter,
+  SelectionItem,
+  SelectionOptions,
+  SelectionStrategy,
+} from '@ts/ui/selection/types';
+
+export default class Selection<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  TItem extends SelectionItem = any,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  TKey extends string | number = any,
+  TDeferred extends boolean = boolean,
+> {
+  options: SelectionOptions<TItem, TKey, TDeferred>;
+
+  _selectionStrategy: SelectionStrategy<TItem, TKey, TDeferred>;
 
   _focusedItemIndex: number;
 
-  _shiftFocusedItemIndex!: number;
+  _shiftFocusedItemIndex?: number;
 
-  constructor(options) {
+  constructor(options: Partial<SelectionOptions<TItem, TKey, TDeferred>>) {
     this.options = extend(this._getDefaultOptions(), options, {
```

Comment:
extend looses typings,
here you can use spread operator to create options objects
and then issues with this.option has type `any` will be solved
```suggestion
    this.options = {
      ...this._getDefaultOptions(),
      ...options,
      selectedItemKeys: options.selectedKeys ?? [],
    };
```

Reply:
fixed in https://github.com/DevExpress/DevExtreme/pull/30768/files#diff-20c3dbd6cf08c2c28dabf97cda749aab84e653438cd15f496c85e6b3370a61fc